src/components/reservation/ReservationForm.jsx

```diff
@@ -0,0 +1,188 @@
+          {
+            date,
+            city,
+            user_id: userId,
+            yacht_id: yachtId,
+          },
+          {
+            headers: {
+              'Content-Type': 'application/json', // Send data as JSON
+            },
+          },
+        );
+
+        if (response.status === 201 || response.status === 200) {
+          const { data } = response;
+          if (data.success) {
+            toast.success(data.message);
+            handleClose();
+            setTimeout(() => {
+              navigate('/reservations');
+              dispatch(fetchReservations({ userId }));
+            }, 1500);
+          } else {
+            toast.error(data.message);
+          }
+        } else {
+          throw new Error(response.statusText);
+        }
+      } catch (error) {
+        toast.error('An error occurred. Please try again later.');
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!yachts.length) {
+      dispatch(fetchYachts());
+    }
+  }, [yachts, dispatch]);
+
+  return (
+    <div className='p-4'>
+      <form onSubmit={handleSubmit} className='max-w-sm mx-auto'>
+        <div className='mb-4'>
+          <label
+            htmlFor='userName'
+            className='block text-gray-700 font-semibold mb-2'
+          >
+            {' '}
+            User Name
+            <input
+              type='text'
+              id='userName'
+              value={user.name}
+              disabled
+              className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+            />
+          </label>
+        </div>
+
+        {yId < 1 ? (
+          <div className='mb-4'>
+            <label
+              htmlFor='yachtName'
+              className='block text-gray-700 font-semibold mb-2'
+            >
+              Yacht Name
+              <input
+                type='text'
+                id='yachtName'
+                value={yachtName}
+                disabled
+                className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+              />
+            </label>
+          </div>
+        ) : (
+          <div className='mb-4'>
+            <label
+              htmlFor='yachtName'
+              className='block text-gray-700 font-semibold mb-2'
+            >
+              Yacht Name
+              <select
+                id='yachtName'
+                value={yachtId}
+                onChange={(e) => setYachtId(e.target.value)}
+                required
+                className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+              >
+                <option value=''>Select Yacht Name</option>
+                {yachts.map((yacht) => (
+                  <option key={yacht.id} value={yacht.id}>
+                    {yacht.model}
+                  </option>
+                ))}
+              </select>
```

Comment:
person should really know what they are reserving, image should be visible so they can be sure it's the right yacht they are reserve. [OPTIONAL]